### PR TITLE
fix(x/ecocredit): basket date criteria and create basket cmd

### PR DESCRIPTION
**Pull request by ryanchristo**
## Description

This pull request fixes basket date criteria, specifically years in the past was not being set when creating a basket, and fixes the create basket CLI command to include years in the past and to correctly parse start date window.

Manual Testing: https://hackmd.io/z8MGG7V5SsaQRf7j5GnrBg

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/ecocredit/basket/client/tx.go [hunk 1]
```diff
@@ -49,9 +50,12 @@ Flags:
 - credit-type-abbrev: filters against credits from this credit type abbreviation (e.g. "BIO").
 - allowed-classes: comma separated (no spaces) list of credit classes allowed to be put in
     the basket (e.g. "C01,C02").
-- min-start-date: the earliest start date for batches of credits allowed into the basket.
+- min-start-date: the earliest start date for batches of credits allowed into the basket
+	(e.g. \"2012-01-01\").
 - start-date-window: the duration of time (in seconds) measured into the past which sets a
```
Comment (clevinson) [inline x/ecocredit/basket/client/tx.go:55]:
```suggestion
- start-date-window: the amount of time (formatted as a duration string) measured into the past which sets a
```

File: x/ecocredit/basket/client/tx.go [hunk 2]
```diff
@@ -97,8 +107,13 @@ Flags:
 				return err
 			}
 
-			if minStartDateString != "" && startDateWindow != 0 {
-				return fmt.Errorf("both %s and %s cannot be set", FlagStartDateWindow, FlagMinimumStartDate)
+			if (minStartDateString != "" && startDateWindow != "") ||
+				(startDateWindow != "" && yearsInThePast != 0) ||
+				(minStartDateString != "" && yearsInThePast != 0) {
```
Comment (clevinson) [inline x/ecocredit/basket/client/tx.go:112]: We should also return an error if all 3 are set
```suggestion
			if (minStartDateString != "" && startDateWindow != "") ||
				(startDateWindow != "" && yearsInThePast != 0) ||
				(minStartDateString != "" && yearsInThePast != 0) ||
				(minStartDateString != "" && startDateWindow != "" && yearsInThePast != 0) {
```

**Review by clevinson — COMMENTED** [review]
Generally looks good. Found one missing case in the CLI checks ensuring only one date criteria is used.

File: x/ecocredit/basket/client/tx.go (hunk 2 above)
Comment (ryanchristo) [reply x/ecocredit/basket/client/tx.go:112]: That's a duplicate check. We don't need this.

Comment (ryanchristo) [reply x/ecocredit/basket/client/tx.go:112]: The added line is already covered with the other checks. We don't need this.

Comment (clevinson) [reply x/ecocredit/basket/client/tx.go:112]: 🤦

**Review by clevinson — APPROVED** [review]

**Pull request merged**